### PR TITLE
Add missing endpoint

### PR DESCRIPTION
**Pull request by gmrukwa**
* missing enpoint `/results/<task-name>` has been added
* Docker image building process has been normalized to the usual
* `schedule` endpoint response is JSON now

### CODE REVIEW

**Comment by gmrukwa** [comment]
[DiviK results endpoint is not implemented yet](https://trello.com/c/hxNzNRIZ/423-divik-results-endpoint-is-not-implemented-yet)

File: .travis.yml [hunk 1]
```diff
@@ -18,17 +20,34 @@ script:
...
               docker push spectreteam/spectre-analysis:beta
-              docker push spectreteam/spectre-analysis:$VERSION-beta
+              docker push spectreteam/spectre-analysis:$MAJOR.$MINOR.$FIXUP-beta
+              docker push spectreteam/spectre-analysis:$MAJOR.$MINOR-beta
+              docker push spectreteam/spectre-analysis:$MAJOR-beta
           fi
-          docker push spectreteam/spectre-analysis:$VERSION-alpha
+          docker push spectreteam/spectre-analysis:$MAJOR.$MINOR.$FIXUP-alpha
+          docker push spectreteam/spectre-analysis:$MAJOR.$MINOR-alpha
+          docker push spectreteam/spectre-analysis:$MAJOR-alpha
```
Comment (mg6) [inline .travis.yml]: Same remarks apply as in https://github.com/spectre-team/divik-worker/pull/4#pullrequestreview-107111593.

File: api.py [hunk 2]
```diff
@@ -29,7 +29,55 @@ def get_algorithms():
 
 def _jsonify(response) -> Response:
     """Pass request data as a response"""
-    return flask.jsonify(response.json()), response.status_code
+    if response.ok:
+        return flask.jsonify(response.json()), response.status_code
+    return "{}", response.status_code
```
Comment (mg6) [inline api.py]: How about informing the user what was the exact problem with the request?

File: api.py [hunk 3]
```diff
@@ -29,7 +29,55 @@ def get_algorithms():
...
+    return list(filter(match, collection))
+
+
+def _select(data, selectors: List[Selector]):
+    for selector in selectors:
+        if isinstance(selector, Matcher):
+            data = _matching(selector, data)[0]
+        else:
+            data = data[selector]
+    return data
```
Comment (mg6) [inline api.py]: `_match_item`, `_matching`, `_select` should be covered by unit tests.

**Review by mg6 — DISMISSED** [review]

File: .travis.yml (hunk 1 above)
Comment (gmrukwa) [reply .travis.yml]: Accepted, with some modifications. `$TRAVIS_REPO_SLUG` is not a name of the docker repo in this case.

File: api.py (hunk 2 above)
Comment (gmrukwa) [reply api.py]: No - this API is public. We want to hide the errors, while developers have access to logs. It is usual approach for web servers due to security reasons.

File: api.py (hunk 3 above)
Comment (gmrukwa) [reply api.py]: Accepted.

**Review by Daroslaw — APPROVED** [review]

**Review by mg6 — APPROVED** [review]

**Pull request merged**